packages/protobuf/src/private/util-common.ts

```diff
@@ -55,18 +54,48 @@ export function makeUtilCommon(): Omit<Util, "newFieldList" | "initFields"> {
               !(val instanceof sourceField.T)
             ) {
               val = new sourceField.T(val);
+            } else if (
+              sourceField &&
+              sourceField.kind === "scalar" &&
+              sourceField.T === ScalarType.BYTES &&
+              !(val instanceof Uint8Array)
+            ) {
+              val = new Uint8Array(val);
             }
             t[localName] = { case: sk, value: val };
             break;
           case "scalar":
           case "enum":
-            t[localName] = s[localName];
+            let target = s[localName];
+            if (member.T === ScalarType.BYTES) {
+              if (!member.repeated && !(target instanceof Uint8Array)) {
+                target = new Uint8Array(target);
+              } else if (member.repeated) {
+                const val = [];
+                for (const v of target) {
+                  val.push(v instanceof Uint8Array ? v : new Uint8Array(v));
+                }
+                target = val;
+              }
+            }
+            t[localName] = target;
```

Comment:
WDYT about introducing a function that converts to Uint8Array if necessary? 


```ts
function toU8Arr(input: ArrayLike<number>): Uint8Array {
  if (input instanceof Uint8Array) {
    return input;
  }
  return new Uint8Array(input);
}
```

This removes the if-else code paths from the call sites. Here, it would allow us to use a ternary operator:

```suggestion
            let target = s[localName];
            if (member.T === ScalarType.BYTES) {
              target = member.repeated ? target.map(toU8Arr) : toU8Arr(target);
            }
            t[localName] = target;
```

I think other call-sites would also benefit, with slightly easier to read code and smaller bundle size.